cms/djangoapps/contentstore/api/views/course_quality.py

```diff
@@ -98,47 +108,48 @@ def _execute_method_and_log_time(func, *args):
         with store.bulk_operations(course_key):
             course = store.get_course(course_key, depth=self._required_course_depth(request, all_requested))
             # Added for EDUCATOR-3660
-            course_key_harward = True if course_key == 'course-v1:HarvardX+SW12.1x+2016' else False
+            course_key_harward = str(course_key) == 'course-v1:HarvardX+SW12.1x+2016'
 
             response = dict(
                 is_self_paced=course.self_paced,
             )
-            if get_bool_param(request, 'sections', all_requested):
-                if course_key_harward:
-                    response.update(
-                        sections=_execute_method_and_log_time(self._sections_quality, course)
-                    )
-                else:
-                    response.update(
-                        sections=self._sections_quality(course)
-                    )
-            if get_bool_param(request, 'subsections', all_requested):
-                if course_key_harward:
-                    response.update(
-                        subsections=_execute_method_and_log_time(self._subsections_quality, course, request)
-                    )
-                else:
-                    response.update(
-                        subsections=self._subsections_quality(course, request)
-                    )
-            if get_bool_param(request, 'units', all_requested):
-                if course_key_harward:
-                    response.update(
-                        units=_execute_method_and_log_time(self._units_quality, course, request)
-                    )
-                else:
-                    response.update(
-                        units=self._units_quality(course, request)
-                    )
-            if get_bool_param(request, 'videos', all_requested):
-                if course_key_harward:
-                    response.update(
-                        videos=_execute_method_and_log_time(self._videos_quality, course)
-                    )
-                else:
-                    response.update(
-                        videos=self._videos_quality(course)
-                    )
+            if not DISABLE_COURSE_CHECKLIST_QUALITY.is_enabled(course_key):
+                if get_bool_param(request, 'sections', all_requested):
+                    if course_key_harward:
```

Comment:
I would suggest to move this part of checking in a central place. There can be two options:
1- Call `response.update(sections=_execute_method_and_log_time)` and decide if we want to log time based on course id condition in `_execute_method_and_log_time` function.
2- Write a wrapper method say `execute_method` and call it like `response.update(sections=execute_method)` and decide which method (with or without logs) to call based on condition.
Computational cost might not be much reduced but we can cut down many number of lines and can make this part of code simpler.